packages/web/server/queries/complex/pools/index.ts

```diff
@@ -57,9 +61,9 @@ export async function getPool(poolId: string): Promise<Pool> {
  *  Params can be used to filter the results by a fuzzy search on the id, type, or coin denoms, as well as a specific id or type. */
 export async function getPools(
   params?: PoolFilter,
-  poolProvider: PoolProvider = getPoolsFromSidecar
+  poolsProvider: PoolsProvider = getPoolsFromSidecar
 ): Promise<Pool[]> {
-  let pools = await poolProvider();
+  let pools = await poolsProvider({ poolIds: params?.poolIds });
```

Comment:
I agree it makes it simpler, however, it's slower than querying the exact pools. Fetching all pools takes ~1.3s, whereas querying with specific IDs completes in ~400ms. Do you think we'll have to add many providers because otherwise it will be a small price to pay for the performance boost?